src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -619,7 +619,23 @@ protected function prepare_item_for_database( $request ) {
 			$changes->post_author = $post_author;
 		}
 
-		return $changes;
+		/**
+		 * Filters a post before it is inserted via the REST API.
+		 *
+		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
+		 *
+		 * Possible hook names include:
+		 *
+		 *  - `rest_pre_insert_wp_template`
+		 *  - `rest_pre_insert_wp_template_part`
+		 *
+		 * @since 6.5.0
+		 *
+		 * @param stdClass        $changes An object representing a single post prepared
+		 *                                 for inserting or updating the database.
+		 * @param WP_REST_Request $request Request object.
+		 */
+		return apply_filters( "rest_pre_insert_{$this->post_type}", $changes, $request );
```

Comment:
This is called on the last line of the `prepare_item_for_database` method, in accordance with `WP_REST_Posts_Controller`:

https://github.com/WordPress/wordpress-develop/blob/6818eb519d07641aeb547bcf6dad9a96784ff161/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1382-L1399

Reply:
In this case, since this is not a new filter, we must not repeat the docblock.